core/src/main/scala/io/chrisdavenport/shellfish/FilesOs.scala

```diff
@@ -0,0 +1,281 @@
+/*
+ * Copyright (c) 2024 Typelevel
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package io.chrisdavenport.shellfish
+
+import cats.syntax.all.*
+import cats.effect.IO
+
+import fs2.{Stream, Chunk}
+import fs2.interop.scodec.*
+import fs2.io.file.{Files, Path, Flags}
+
+import scodec.{Codec, codecs}
+import scodec.bits.ByteVector
+
+object FilesOs {
+
+  val files = Files[IO]
+
+  // Read operations:
+
+  /**
+   * Reads the contents of the file at the path and returns it as a String.
+   * @param path
+   *   The path to read from
+   * @return
+   *   The file loaded in memory as a String
+   */
+  def read(path: Path): IO[String] = files.readUtf8(path).compile.string
```

Comment:
I will specify that we're reading the string as a UTF-8 in the name of the method also, because despite being a reasonable default it's not the only way of reading a string.

```suggestion
  def readUtf8(path: Path): IO[String] = files.readUtf8(path).compile.string
```

I would also add a `read(charset: Charset)(path: Path)` method or similar to cover the other cases.

Reply:
Oh why not just use in that case the `readAs` method? It has a codec that accepts a charset

```scala
for
  file <- path.readAs(using string(StandardCharsets.UTF_8))
yield ()

Reply:
I personally prefer
```scala
def read(path: Path): IO[String]
def read(path: Path, charset: StandardCharsets): IO[String]
// or just
def read(path: Path, charset: StandardCharsets = UTF_8): IO[String]
```

But maybe it's good to be consistent with fs2 naming.

Reply:
> But maybe it's good to be consistent with fs2 naming.
This specifically. Also, scodec is not really made to read and/or manipulate strings.

Reply:
Aside: I'd avoid default arguments in public APIs as it ends up hard to evolve the API over time.

Reply:
Hmm I personally don't like adding `Utf8` to the standalone methods because most of the time the users are going to read it as utf-8 anyways (I think that's why in `os-lib` they leaved it that way as well as in Python, so I think we should follow that approach). There are also four more character that the user has to write in order to do the most common operations (one of them being very far away from the home row in the keyboard) 🤔

Another problem on that is to deal with beginners, as most of the time they will not know or worry about encodings or charsets and just want to load the file in memory, and as Rishad told me most of them don't read the documentation 😂 so, if I were to put myself in the shoes of someone new to the library or programming in general and see that to "read" a file it is necessary to pass a charset, and in order to use the one he probably needs he has to add this `utf8` thing at the end to load it as a string? At that point I will be very confused 🤣 that's why I prefer the one Thanh suggested using overrides.

BUT, what do you guys think if I add both of them? And leave `read`, `write` and `append` as alias defaults? Something like: 

```scala
def read(path: Path): IO[String] = readUtf8(path)
def read(charset: Charset)(path: Path):  IO[String] = ???
def readUtf8(path: Path): ???

...    // Same with write and append
```

In that way, more experienced users that use custom charsets can see that the standalone methods defaults to UTF8 and use the `Utf8` variants whenever they are working with multiple encodings, and leave the rest use the defaults?

Reply:
I'll sound like a broken record but we shouldn't offer multiple ways to do the same thing so I'm 👎 on aliasing. Maybe don't offer a read method at all and have `readString(path: Path)` and `readStringWithCharset(path: Path, charset: Charset)`.

Reply:
Oh, in that case we can mention in the scaladocs that `read` defaults to UTF-8 and create a `readWithCharset` like you said, what do you think?

Reply:
I do agree, let's do what Micheal suggests